app/routes/task_routes.py

```diff
@@ -1 +1,85 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from app.models.task import Task
+from .route_utilities import validate_model, chat_post_message, create_model
+from sqlalchemy import asc, desc
+from ..db import db
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+@tasks_bp.post("")
+def create_task():
+    request_body = request.get_json()
+    return create_model(Task, request_body, "task")
+
+
+@tasks_bp.get("")
+def get_all_tasks():
+    query = db.select(Task)
+    
+    title_param = request.args.get("title")
+    if title_param:
+        query = query.where(Task.title.ilike(f"%{title_param}%"))
+
+    sort_param = request.args.get("sort")
+    if sort_param:
+        if sort_param.lower() == "asc":
+            query = query.order_by(asc(Task.title))
+        elif sort_param.lower() == "desc":
+            query = query.order_by(desc(Task.title))
+
+    query = query.order_by(Task.id)
+
+    tasks = db.session.scalars(query)
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+    return tasks_response
+
+
+@tasks_bp.get("/<task_id>")
+def get_one_task(task_id):
+    task = validate_model(Task, task_id)
+
+    return {"task": task.to_dict()}
+
+
+@tasks_bp.put("/<task_id>")
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
This is another opportunity to D.R.Y. up our code! Notice how in this route and the `PUT` route for `goal_routes.py` we follow the pattern of:
```py
object.ATTRIBUTE = request_body["ATTRIBUTE"]
```
We use `hasattr` and [setattr](https://docs.python.org/3/library/functions.html#setattr) to make a helper function to update our `Task` and `Goal` model. It would look like this: 
```py
def update_model(obj, data):
    for attr, value in data.items():
        if hasattr(obj, attr):
            setattr(obj, attr, value)
    
    db.session.commit()

    return Response(status=204, mimetype="application/json")
```

This refactor not only makes our code D.R.Y but shows that we recognize logic that has higher level usability while handling cases of keys not being found!